tests/e2e/kubernetes/kubernetes_test.go

```diff
@@ -539,3 +704,270 @@ func waitPodDeletion(done, podsDeleted chan struct{}, t *testing.T) {
 		time.Sleep(15 * time.Second)
 	}
 }
+
+func waitAllPodsRunning(t *testing.T, namespace string, done, podsRunning chan struct{}) {
+	for {
+		select {
+		case <-done: // if timeout was reached
```

Comment:
What about use context and then use ctx.Done()?

Reply:
I can do that in a follow up PR if needed.